### PR TITLE
Implemented optional pre-selected rows

### PR DESCRIPTION
**Pull request by ognyan-kostadinov**
**Component**:

**Description**:

**Design**:

**Breaking Changes**:

- [] Breaking Changes

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: src/lib/components/tablev2/Tablev2.component.tsx [hunk 1]
```diff
@@ -44,6 +45,7 @@ export type TableProps<
   onBottom?: (rowLength: number) => void;
   onBottomOffset?: number;
   allFilters?: { id: string; value: string }[];
+  initiallySelectedRows?: Set<string>;
```
Comment (JBWatenbergScality) [inline src/lib/components/tablev2/Tablev2.component.tsx]:
```suggestion
  initiallySelectedRowsIds?: Set<string>;
```

File: src/lib/components/tablev2/Tablev2.component.tsx [hunk 2]
```diff
@@ -117,6 +120,18 @@ function Table<
...
+    if (initiallySelectedRows) {
+      return Array.from(initiallySelectedRows).reduce(
+        (accumulatedValue, currentValue) => (
+          (accumulatedValue[currentValue] = true), accumulatedValue
+        ),
+        {},
+      );
+    }
+    return {};
+  }, [initiallySelectedRows]) as Record<IdType<DATA_ROW>, boolean>;
```
Comment (JBWatenbergScality) [inline src/lib/components/tablev2/Tablev2.component.tsx]: initiallySelectedRows being an object this  useMemo will get executed on every render.

Comment (JBWatenbergScality) [inline src/lib/components/tablev2/Tablev2.component.tsx]: I think this type cast could be removed if the reduce function implementation was reviewed as below:

```
Array.from(initiallySelectedRows).reduce(
        (accumulatedValue, currentValue) => (
          ({...accumulatedValue, [currentValue]: true}), accumulatedValue
        ),
        {},
      );
```

Comment (ognyan-kostadinov) [reply src/lib/components/tablev2/Tablev2.component.tsx]: Unfortunately it's not the case, it still considers it `{}` if not explicitly typecasted

File: src/lib/components/tablev2/Tablev2.component.tsx [hunk 3]
```diff
@@ -117,6 +120,19 @@ function Table<
...
+    if (initiallySelectedRowsIds) {
+      return Array.from(initiallySelectedRowsIds).reduce(
+        (accumulatedValue, currentValue) => ({
+          ...accumulatedValue,
+          [currentValue]: true,
+        }),
+        {},
+      );
+    }
+    return {};
```
Comment (JBWatenbergScality) [inline src/lib/components/tablev2/Tablev2.component.tsx:133]:
```suggestion
  const formattedInitiallySelectedRows = React.useMemo(() => {
    
      return Array.from(initiallySelectedRowsIds || []).reduce(
        (accumulatedValue, currentValue) => ({
          ...accumulatedValue,
          [currentValue]: true,
        }),
        {},
      );
```

**Review by JBWatenbergScality — APPROVED** [review]

**Pull request merged**